frontend_lib/src/appContentFactory.js

```diff
@@ -353,16 +365,21 @@ export function appContentFactory (WrappedComponent) {
           created_raw: revision.created,
           created: displayDistanceDate(revision.created, loggedUser.lang),
           timelineType: 'revision',
-          commentList: revision.comment_ids.map(ci => ({
-            timelineType: 'comment',
-            ...resCommentWithProperDate.find(c => c.content_id === ci)
-          })),
+          commentList: revision.comment_ids.map(ci => (
+            timelineCommentList.find(c => c.content_id === ci)
+          )),
           number: i + 1,
           hasBeenRead: true
         }))
         .flatMap(revision => [revision, ...revision.commentList])
     }
 
+    replaceComment = (comment, timeline) => {
+      return timeline.map(
+        item => item.timelineType === 'comment' && item.content_id === comment.content_id ? comment : item
```

Comment:
don't we have a constant for 'comment'?

Reply:
In this context (`timelineType`) no, I can add it, this doesn't seem to be too invasive.

Reply:
In fact there is one, but the code from here must have been written before its creation.

Reply:
Corrected.